### ISSUE TITLE
Ujednolicimy dane metod dostawy

### ISSUE THREAD

**Issue by PawelTaberski** [allegro/allegro-api#1135]
- 4.02.2019 ujednolicimy nazwy metod dostawy w REST API i WebAPI. Dzięki temu:
   - w całym API Allegro będziemy posługiwali się tymi samymi nazwami metod dostawy
   - nazwy będą lepiej się prezentować w aplikacjach mobilnych, ponieważ wiele z nich skrócimy
- 28.02.2019 zmienimy maksymalne wartości dla metod wysyłki w REST API i WebAPI.
   - Zmianę wprowadzamy, aby ujednolicić wartości maksymalne dla metod wysyłki w API Allegro.

[Więcej informacji.](https://developer.allegro.pl/news/2019-01-30-metody_dostawy/)

**Comment by Onixarts** [comment]
Wspaniale. Bo przy tylu dostawach to się można mocno pogubić. Zwłaszcza, że część z nich ma słowo pobranie, inne pobrania nie mają pobrania, z kolei płatne z góry raz są po przedpłacie raz nie ;).

Jeśli mogę dorzucić moją sugestię, to w mojej ocenie przesyłka powinna być **XXXXX** oraz **XXXXX pobranie**. W programach i w mowie potocznej zazwyczaj funkcjonuje przesyłka zwykła i pobraniowa, nie żadna przedpłacona, po przedpłacie itp - przedpłata, płatność z góry jest w zasadzie domyślna.

**Comment by PawelTaberski** [comment]
@Onixarts Przekazałem Twoją sugestię, pamiętaj że w REST API zawsze masz pole "paymentPolicy" gdzie określamy typ płatności. Oczywiście rozumiem, ze Twoja sugestia dotyczy samej nazwy i tak przekazałem 👍

**Comment by manhunto** [comment]
Czy w jakiś sposób można pobrać dopuszczalne wartości dla konkretnej metody wysyłki za pomocą REST API? Chodzi mi np o maksymalną cenę. http://q.i-systems.pl/file/afb83a67.png 
Odpowiedź z api wygląda mniej więcej tak, ale chciałbym wprowadzić walidację po swojej stronie.
`{"errors":[{"code":"INVALID_FIRST_ITEM_PRICE","message":"Invalid first item price (min: 0.00, max: 8.99)","details":null (truncated...)`
Albo to że nie możemy ustawić ceny następnego przedmiotu
`{"errors":[{"code":"INVALID_NEXT_ITEM_PRICE","message":"Invalid next item price (min: 0.00, max: 0.00)","details":null," (truncated...)`

Z tego co kojarze to w WebApi była taka informacja, ale w rest nie mogę czegoś takiego znaleźć

**Comment by alekskuc** [comment]
Witam,
Zamiana nazw metod dostaw została opisana następująco (na pierwszym przykładzie):

REST API ID	Aktualna nazwa w REST API	WebAPI ID	Aktualna nazwa w WebAPI	Nowa nazwa
85c3ad2f-4ec1-446c-866e-63473ed10e26	Allegro Kurier24 InPost	18	Allegro Kurier24 InPost (za pobraniem)	Allegro Kurier24 InPost pobranie

Co dokładnie się zmieni w nazwach przysyłanych z WEB API?

Allegro Kurier24 InPost (pierwsza sztuka)
Allegro Kurier24 InPost (kolejna sztuka)
Allegro Kurier24 InPost (ilość w paczce)

Allegro Kurier24 InPost (za pobraniem) (pierwsza sztuka)
Allegro Kurier24 InPost (za pobraniem) (kolejna sztuka)
Allegro Kurier24 InPost (za pobraniem) (ilość w paczce)

Samej nazwy „Allegro Kurier24 InPost (za pobraniem)” bez dodatkowych informacji w SOAP WEB API nie ma.

**Comment by PawelTaberski** [comment]
@manhunto Obecnie taką informacje zwracamy tylko jak podasz wartość spoza zakresu, i nigdy nie pokazujemy maksymalnych wartości. Ale może rzeczywiście powinniśmy ją pokazywać, może np. w zasobie GET  /sale/delivery-methods. Zgłosiłem taką sugestię, chyba że widziałbyś inne wygodniejsze dla Ciebie rozwiązanie.
 @alekskuc W tym przypadku dojdzie na końcu zapis pobranie, ostatecznie prezentowana nazwa tej metody dostawy będzie:
Allegro Kurier24 InPost pobranie
Dodatkowy nawias z już nie tyle samą nazwą metody co nawiązaniu do zawartości fid-a nie powinien się zmienić tzn. zostanie (pierwsza sztuka), (kolejna sztuka) itd.

**Comment by manhunto** [comment]
@PawelTaberski tak było by to bardzo pomocne :) Metoda GET /sale/delivery-methods wydaje się dobrym miejscem na to, ale dostosuję się jeśli będzie to nowy zasób czy cokolwiek innego

**Comment by pskz** [comment]
Zgadzam się z @manhunto, obecnie te zakresy można uzyskać tylko przez pobranie fidów do kategorii, korzystając z WebApi. W REST wysyłki przekazujemy osobno jako cennik dostaw, więc nagle zniknęła możliwość sprawdzenia tych danych po stronie aplikacji, przed ich wysłaniem.

**Comment by PawelTaberski** [comment]
@pskz Tak, jak pisałem przekazałem sugestię, jeśli zdecydujemy się wdrożyć takie rozwiązanie, to oczywiście damy dedykowany komunikat.
